packages/@aws-cdk/aws-events-targets/lib/api-gateway.ts

```diff
@@ -74,7 +74,7 @@ export interface ApiGatewayProps extends TargetBaseProps {
  */
 export class ApiGateway implements events.IRuleTarget {
 
-  constructor(public readonly restApi: api.RestApi, private readonly props?: ApiGatewayProps) {
+  constructor(public readonly restApi: api.IRestApi, private readonly props?: ApiGatewayProps) {
```

Comment:
The error is:

```
API elements with incompatible changes:
err  - PROP @aws-cdk/aws-events-targets.ApiGateway.restApi: type @aws-cdk/aws-apigateway.IRestApi (formerly @aws-cdk/aws-apigateway.RestApi): @aws-cdk/aws-apigateway.IRestApi does not extend @aws-cdk/aws-apigateway.RestApi [changed-type:@aws-cdk/aws-events-targets.ApiGateway.restApi]
```

The reason why this change is incompatible is because `public readonly restApi` declares both an INPUT parameter and an OUTPUT parameter, and we can only change one in one direction, and the other in the other. That makes it somewhat more complex, so It probably needs to look somewhat like this:

```
private readonly _restApi: IRestApi;

constructor(restApi: IRestApi, private readonly props?: ApiGatewayProps) {
  this._restApi = resetApi;
}

public get restApi(): RestApi {
  if (!(this._restApi instanceof RestApi)) {
    throw new Error('The RestApi is not a RestApi construct, and cannot be retrieved this way');
  }
  return this._restApi;
}
```